.github/workflows/tests.yml

```diff
@@ -11,33 +10,16 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        os: [ 'ubuntu:16.04', 'ubuntu:20.04', 'ubuntu:22.04' ]
+        os: [ 'ubuntu:18.04', 'ubuntu:20.04', 'ubuntu:22.04' ]
```

Comment:
Why?
16.04 is used because of the older openssl version - we keep to have coverage for that version

Reply:
i was working based off the same line of reasoning laid out in #1397 , but also a mistaken assumption that the code reuse necessary to support 16.04 would completely undermine taking out the action in the first place

as it turns out, that's significantly easier than i thought, so 16.04 support has been re-added